v4/weaviate/connection/rest.go

```diff
@@ -27,12 +27,40 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 	if client == nil {
 		client = &http.Client{}
 	}
-
-	return &Connection{
+	connection := &Connection{
 		basePath:   scheme + "://" + host + "/" + apiVersion,
 		httpClient: client,
 		headers:    headers,
 	}
+
+	// background goroutine that periodically refreshes the auth token.
+	// The oauth2 package only refreshes the Tokens on new http requests => if there is no request for the lifetime of
+	// the refresh token the client will become de-authenticated without this.
+	go func(con *Connection) {
+		transport, ok := con.httpClient.Transport.(*oauth2.Transport)
+		if !ok {
+			return
+		}
+		for {
+			token, err := transport.Source.Token()
+			if err != nil {
+				return
+			}
+			// there is no point in manual refreshing if there is no refresh token
+			if token.RefreshToken == "" {
+				return
+			}
+
+			timeToSleep := token.Expiry.Sub(time.Now()) - time.Second/10
+			time.Sleep(timeToSleep)
+			_, err = connection.RunREST(context.TODO(), "/meta", http.MethodGet, nil)
+			if err != nil {
+				return
+			}
```

Comment:
Maybe we would want to log or propagate these errors

Reply:
added logging

Reply:
Hmm I'm not seeing any logging.  What I mean is that the goroutine should be logging on errors. So `if err != nil`, log the error message before returning from the goroutine

Reply:
There definitely is logging, are you loocking at the [new code](https://github.com/weaviate/weaviate-go-client/blob/b35f005a054992a48f950ab554b36d01bdc24be4/v4/weaviate/connection/rest.go#L84)? The snippet here is not updated

Reply:
Ah weird, I was looking at the refreshed changes [here](https://github.com/weaviate/weaviate-go-client/pull/101/files), but I guess they were still outdated. Ok LGTM